source/MaterialXCore/Document.h

```diff
@@ -680,6 +708,8 @@ class MX_CORE_API Document : public GraphElement
   private:
     class Cache;
     std::unique_ptr<Cache> _cache;
+    // Data library for the document
+    ConstDocumentPtr _dataLibrary;
```

Comment:
I think other methods like `initialize()` and `copy()` probably also need to be updated to handle this extra member variable.